tests/pytest_tests/unit_tests/test_lib/test_filenames.py

```diff
@@ -0,0 +1,41 @@
+import tempfile
+
+import pytest
+from wandb.sdk.lib.filenames import exclude_wandb_fn, filtered_dir
+
+
+def test_filtered_dir_one_parameter():
+    with tempfile.TemporaryDirectory() as tempdir:
+        with open(tempdir + "/foo.txt", "w") as f:
+            f.write("test")
+        filtered_dir(tempdir, lambda path: True, lambda path: False)
+
+
+def test_filtered_dir_two_parameters():
+    with tempfile.TemporaryDirectory() as tempdir:
+        with open(tempdir + "/foo.txt", "w") as f:
+            f.write("test")
+        filtered_dir(tempdir, lambda path, root: True, lambda path, root: False)
+
+
+@pytest.mark.parametrize(
```

Comment:
this test summarizes the change best:

1. (no change) by default exclude files under `<root>/wandb` or `<root>/.wandb`
2. (new) by default DO NOT exclude files under directories named `wandb` or `.wandb` that are not direct children of `<root>`
3. (new) by default DO NOT exclude files just because `wandb/` or `.wandb/` is somewhere in the absolute file path.

Reply:
We might want to notify users of this change, there are likely some **very** large `wandb` folders in filepaths that are currently not creating jobs, but would after this change. When intentionally trying to create a job, this change is only good, but it feels likely that this could have a big unintended impact. It would look like to users that after they updated the sdk, suddenly none of their runs finish, as they all hang trying to upload massive `wandb` folders to artifacts. 

Just a thought, not really sure how to remedy.

Reply:
the default `include_fn` is:

```
def _is_py_path(path: str) -> bool:
    return path.endswith(".py")
```

so in most cases this would only pick up python scripts outside of `<root>/.wandb` and `<root>/wandb`

i think in the vast majority of cases this will fix a lot of silently and mysteriously broken calls to `log_code`, but we're definitely also going to notify users of this change.

Reply:
more generally, i feel like the only real bug here was that `wandb` or `.wandb` *ANYWHERE* in the path was broken, but i can imagine in some cases like sandboxed FS where your path prefix is set, this would be completely broken and fail silently.  I also believe that excluding by default, e.g., `<root>/a/b/c/wandb/*` is incorrect, as it seems very likely that a project might have architected their code to encapsulate all uses of `wandb` to live inside a single directory.